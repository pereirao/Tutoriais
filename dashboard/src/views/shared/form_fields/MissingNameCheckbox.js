// @flow

import React, {PureComponent} from 'react';
import {Checkbox} from 'semantic-ui-react';
import style_sheet from './form_fields.module.css';

type MissingNameCheckboxProps = { onChangeText: (string, Object) => void };

class MissingNameCheckbox extends PureComponent {
  props: MissingNameCheckboxProps

  static defaultProps = {name: 'missing_name', labelEl: null}
  input: ReactClass<*>

  onChange = (_e, {checked}) => {
    this.props.onChangeText(checked, true);
  }

  render() {
    const {label} = this.props;
    return (
      <Checkbox className={style_sheet.checkbox} label={label} onChange={this.onChange}/>
    );
  }
}

export default MissingNameCheckbox;
