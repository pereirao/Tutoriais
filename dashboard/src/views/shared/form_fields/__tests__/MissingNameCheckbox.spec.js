import React from 'react';

import MissingNameCheckbox from '../MissingNameCheckbox';

it('renders correctly', () => {
  expect(shallow(<MissingNameCheckbox />)).toMatchSnapshot();
});
