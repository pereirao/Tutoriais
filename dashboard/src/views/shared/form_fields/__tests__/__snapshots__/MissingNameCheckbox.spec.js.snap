// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Checkbox
  className="checkbox"
  onChange={[Function]}
  type="checkbox"
/>
`;
